File: backend/backend/Models/Vaccination.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Vaccination
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required, MaxLength(100)]
        public string Name { get; set; }

        [Required]
        public DateTime Date { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        [ForeignKey("Animal")]
        public Guid AnimalId { get; set; }
        public Animal Animal { get; set; }
    }
}

File: frontend/client_app/lib/models/vaccination_models/vaccination.dart
class Vaccination {
  final String id;
  final String name;
  final DateTime date;

  Vaccination({
    required this.id,
    required this.name,
    required this.date,
  });

  factory Vaccination.fromJson(Map<String, dynamic> json) {
    return Vaccination(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      date: json['date'] != null
          ? DateTime.tryParse(json['date']) ?? DateTime.now()
          : DateTime.now(),
    );
  }
}
